superset-frontend/src/datasource/DatasourceEditor.jsx

```diff
@@ -842,10 +842,15 @@ class DatasourceEditor extends React.PureComponent {
         {this.allowEditSource && (
           <EditLockContainer>
             <span role="button" tabIndex={0} onClick={this.onChangeEditMode}>
-              <Icon
-                color={supersetTheme.colors.grayscale.base}
-                name={this.state.isEditMode ? 'lock-unlocked' : 'lock-locked'}
-              />
+              {this.state.isEditMode ? (
+                <Icons.LockUnlocked
+                  iconColor={supersetTheme.colors.grayscale.base}
```

Comment:
Can you change to use `withTheme` or `useTheme` to reference the `theme` object?

Reply:
I'm going to do this in a follow up and get rid of all the supersetThemes being used in components.